packages/grid/src/vaadin-grid-column-group.js

```diff
@@ -205,6 +204,17 @@ class GridColumnGroup extends ColumnBaseMixin(PolymerElement) {
   /** @private */
   _updateVisibleChildColumns(childColumns) {
     this._visibleChildColumns = Array.prototype.filter.call(childColumns, (col) => !col.hidden);
+    this._colSpan = this._visibleChildColumns.length;
+
+    if (!this._ignoreVisibleChildColumns) {
+      if (this._visibleChildColumns.length === 0) {
+        if (!this.hidden) {
```

Comment:
Added this condition, which fixes another issue - we don't want to set `this._autoHidden` to true if the group was already hidden.